File: livekit_server/agent-starter-python/list_models.py
import asyncio
import os
from dotenv import load_dotenv
from google import genai

load_dotenv(".env.local")

def test():
    client = genai.Client(api_key=os.getenv("GOOGLE_API_KEY"))
    for model in client.models.list():
        print(model.name)

test()

File: marketing-server/list_models.ts
import { GoogleGenAI } from '@google/genai';
import * as dotenv from 'dotenv';
import path from 'path';

dotenv.config({ path: path.join(process.cwd(), '..', '.env') });
const ai = new GoogleGenAI({ apiKey: process.env.GEMINI_API_KEY });

async function run() {
    try {
        const response = await ai.models.list();
        for await (const model of response) {
            if (model.name.includes("image") || model.name.includes("gen") || model.name.includes("vision")) {
                console.log(model.name);
            }
        }
    } catch (e) {
        console.error("Error:", e.message);
    }
}
run();
